pinot-controller/src/main/java/org/apache/pinot/controller/api/upload/ZKOperator.java

```diff
@@ -60,8 +62,9 @@ public ZKOperator(PinotHelixResourceManager pinotHelixResourceManager, Controlle
   }
 
   public void completeSegmentOperations(String tableNameWithType, SegmentMetadata segmentMetadata,
-      @Nullable URI finalSegmentLocationURI, File segmentFile, String downloadUrl, @Nullable String crypterName,
-      long segmentSizeInBytes, boolean enableParallelPushProtection, boolean allowRefresh, HttpHeaders headers)
+      FileUploadType uploadType, @Nullable URI finalSegmentLocationURI, File segmentFile, String sourceDownloadURIStr,
```

Comment:
Suggest not introducing upload type into this module. When it is configured to copy the segment to the deep store, we can check if `sourceDownloadURIStr` is provided, and copy accordingly.

Reply:
Addressed all other comments. For this one, I kinda prefer keeping the uploadType, so it is very clear when reading. Don't want to rely on `sourceDownloadURIStr` as it can be non-null even in URI push case, but we want to use segment file in that case.